src/scripts/main.js

```diff
@@ -1,3 +1,218 @@
+      i === 2
+        ? document.createElement('select')
+        : document.createElement('input');
+
+    if (i === 3 || i === 4) {
+      elemToAdd.setAttribute('type', 'number');
+    } else if (elemToAdd.tagName === 'INPUT') {
+      elemToAdd.setAttribute('type', 'text');
+    }
+
+    elemToAdd.setAttribute('name', input);
+    elemToAdd.setAttribute('data-qa', input);
+
+    const lblTextContent = input.charAt(0).toUpperCase() + input.slice(1);
+
+    label.textContent = `${lblTextContent}:`;
+    label.append(elemToAdd);
+    form.append(label);
+  });
+
+  button.textContent = 'Save to table';
+  form.append(button);
+  body.append(form);
+  createOptionsSelect();
+};
+
+createForm();
+
+// #region listeners
+// Сортування таблиці
+thead.addEventListener('click', (e) => {
+  const rows = [...document.querySelectorAll('tbody tr')];
+  const headClicked = e.target.textContent;
+  const indexHead = headings.indexOf(headClicked);
+
+  if (prevClicked !== headClicked) {
+    clickCounter = 0;
+  }
+
+  clickCounter++;
+
+  const sortedRow = rows.sort((elem1, elem2) => {
+    const value1 = elem1.children[indexHead].textContent;
+    const value2 = elem2.children[indexHead].textContent;
+    const howToSort = clickCounter % 2 !== 0;
+
+    if (indexHead === 3 || indexHead === 4) {
+      const num1 = indexHead === 4 ? +value1.replace(/[^0-9]+/g, '') : +value1;
+      const num2 = indexHead === 4 ? +value2.replace(/[^0-9]+/g, '') : +value2;
+
+      return howToSort ? num1 - num2 : num2 - num1;
+    }
+
+    return howToSort
+      ? value1.localeCompare(value2)
+      : value2.localeCompare(value1);
+  });
+
+  tbody.innerHTML = '';
+  sortedRow.forEach((row) => tbody.append(row));
+  prevClicked = headClicked;
+});
+
+tbody.addEventListener('click', (e) => {
+  const selectedElem = e.target;
+  const rows = [...document.querySelectorAll('tbody tr')];
+
+  if (prevRow !== undefined) {
+    prevRow.classList.remove('active');
+  }
+
+  const row = rows.find((elem) => {
+    const childrenOfRow = [...elem.children];
+
+    return childrenOfRow[childrenOfRow.indexOf(selectedElem)];
+  });
+
+  row.classList.toggle('active');
+  prevRow = row;
+});
+
+tbody.addEventListener('dblclick', (e) => {
+  const selectedElem = e.target;
+  const oldText = e.target.textContent;
+  const newValue = document.createElement('input');
+
+  newValue.classList.add('cell-input');
+  newValue.value = selectedElem.textContent;
+
+  const row = selectedElem.closest('tr');
+
+  if (row) {
+    selectedElem.textContent = '';
+    selectedElem.appendChild(newValue);
+    newValue.focus();
+
+    ['blur', 'keydown'].forEach((value) => {
+      newValue.addEventListener(value, (keyPress) => {
+        if (
+          value === 'blur' ||
+          (value === 'keydown' && keyPress.key === 'Enter')
+        ) {
+          selectedElem.textContent =
+            newValue.value === '' ? oldText : newValue.value;
+        }
+      });
+    });
+  }
+});
```

Comment:
The implementation for editing table cells by double-clicking is mostly correct, but there is a minor issue. The task description specifies that only one cell can be edited at a time. Ensure that if another cell is being edited, it should be saved or reverted before allowing a new cell to be edited. This can be handled by checking if an input already exists and handling it appropriately before creating a new input.